src/api.js

```diff
@@ -1,16 +1,34 @@
+import { logError } from '@edx/frontend-platform/logging';
 import { examRequiresAccessToken, store } from './data';
 
 export function isExam() {
+  console.log("store: ", store);
+  console.log("state: ", store.getState());
+  console.log("exam state", store.getState().examState);
   const { exam } = store.getState().examState;
+  if (!exam) {
+    return false;
+  }
   return exam.id !== null;
 }
 
 export function getExamAccess() {
-  const { examAccessToken } = store.getState().examState;
+  const { exam, examAccessToken } = store.getState().examState;
+  console.log("state impl", store.getState().examState);
+  if (!exam) {
+    logError('Failed to get exam access token. Not an exam.');
+    return '';
+  }
   return examAccessToken.exam_access_token;
 }
 
 export async function fetchExamAccess() {
-  const { dispatch } = store;
+  console.log("exam state", store.getState().examState);
+  const { dispatch, exam } = store;
+  if (!exam) {
+    return new Promise((resolve, reject) => {
```

Comment:
That does make sense to me. The only issue I was running into earlier which is what lead me down this path is, in the case where there is an exam and the access token thunk is dispatched, a promise is returned. So wouldn't a promise also need to be returned in the case where there is no exam? For example, I did get an error when I tried to just `return;` for getExamAccess, since it was inconsistent with the return of an exam access string.

Reply:
the return types should be consistent so I'd agree this should always return a promise. That said, we don't really need the error so a simple `return Promise.resolve()` is likely sufficient.

Reply:
Okay, I'll tinker with that!

Reply:
Should I remove the error for getExamAccess above? Since the error isn't really needed? It's just handling the case where there is no exam?